ow_power_system/src/power_system_node.cpp

```diff
@@ -17,8 +16,8 @@ using namespace std::chrono;
 void PowerSystemNode::powerCallback(const std_msgs::Float64::ConstPtr& msg)
 {
   // Set mechanical power value to rostopic subscription
-  double mechanical_power = msg->data;  // [W]
-
+  double mechanical_power = checkForFaults(msg->data);  // [W]
+  
   // Temperature estimate based on pseudorandom noise and fixed range
   double temperature_estimate =
       m_min_temperature + static_cast<double>(rand()) / RAND_MAX * (m_max_temperature - m_min_temperature);
```

Comment:
Minor comment: contrast with FaultInjector::getRandomFloatFromRange, assuming they do the same thing except use double vs float.   Factoring possible but not important.

Reply:
Ussama/Jason wrote that separately from the Fault work, but yea we could refactor this in the future.